development/build/index.js

```diff
@@ -164,89 +166,110 @@ function defineAndRunBuildTasks() {
+          type: 'boolean',
+        })
+        .option('build-type', {
+          default: BuildType.main,
+          description: 'The type of build to create.',
+          choices: Object.keys(BuildType),
+        })
+        .option('build-version', {
+          default: 0,
+          description:
+            'The build version. This is set only for non-main build types. The build version is used in the "prerelease" segment of the extension version, e.g. `[major].[minor].[patch]-[build-type].[build-version]``',
+          type: 'number',
+        })
+        .option('lint-fence-files', {
+          description:
+            'Whether files with code fences should be linted after fences have been removed. The build will fail if linting fails. This defaults to `false` if the entry task is `dev` or `testDev`. Otherwise this defaults to `true`.',
+          type: 'boolean',
+        })
+        .option('lockdown', {
+          default: true,
+          description:
+            'Whether to include SES lockdown files in the extension bundle. Setting this to `false` can be useful during development if you want to handle lockdown errors later.',
+          type: 'boolean',
+        })
+        .option('policy-only', {
+          default: false,
+          description:
+            'Stop the build after generating the LavaMoat policy, skipping any writes to disk other than the LavaMoat policy itself.',
+          type: 'boolean',
+        })
+        .option('skip-stats', {
+          default: false,
+          description:
+            'Whether to skip logging the time to completion for each task to the console. This is meant primarily for internal use, to prevent duplicate logging.',
+          hidden: true,
+          type: 'boolean',
+        })
+        .check((args) => {
+          if (!Number.isInteger(args.buildVersion)) {
+            throw new Error(
+              `Expected integer for 'build-version', got '${args.buildVersion}'`,
+            );
+          } else if (!Object.values(TASKS).includes(args.task)) {
+            throw new Error(`Invalid task: '${args.task}'`);
+          }
+          return true;
+        }),
+    )
+    .help('help');
 
-  // Manually default this to `false` for dev builds only.
-  const shouldLintFenceFiles = process.argv.includes(
-    `--${NamedArgs.LintFenceFiles}`,
-  )
-    ? argv[NamedArgs.LintFenceFiles]
-    : !/dev/iu.test(entryTask);
+  const {
+    applyLavamoat: applyLavaMoat,
+    buildType,
+    buildVersion,
+    lintFenceFiles,
+    lockdown,
+    policyOnly,
+    skipStats,
+    task,
+  } = argv;
 
-  const policyOnly = argv[NamedArgs.PolicyOnly];
+  let shouldLintFenceFiles = lintFenceFiles;
+  if (shouldLintFenceFiles === undefined) {
+    // Manually default this to `false` for dev builds only.
+    shouldLintFenceFiles = !/dev/iu.test(task);
+  }
```

Comment:
Would this work too?

``` suggestion
  // Manually default this to `false` for dev builds only.
  const shouldLintFenceFiles = lintFenceFiles ?? !/dev/iu.test(task)
```

Reply:
Yes, good idea, I forgot about that syntax in the moment.